pickfirstleaf: fix bug in address de-duplication

arjan-bal opened the pull request:
Due to a bug in the new pickfirst balancer, it wasn't de-duplicating addresses in the resolver update. The only user visible impact of this seems to be less frequent picker updates after the first pass in happy eyeballs and incorrect interleaving of IPv4/IPv6 addresses during the first happy eyeballs pass.

RELEASE NOTES:
* balancer/pickfirst: Fix a bug where duplicate addresses were not being ignored as intended.

easwars reviewed (approved):
The release note doesn't seem to indicate that we are actually fixing a bug. It looks more like we are adding the logic to dedup addresses. Although technically this PR is doing that since we weren't deduping before :)

arjan-bal commented:
> This part of the description is very confusing: `The only user visible impact of this seems to be more less frequent picker updates`

Fixed. It should have been "less frequent picker updates" since the address list would be longer. I changed the phrasing of the sentence midway through typing it.

> The release note doesn't seem to indicate that we are actually fixing a bug. It looks more like we are adding the logic to dedup addresses. Although technically this PR is doing that since we weren't deduping before :)

Updated the release notes to mention that not ignoring duplicate addresses was a bug and it is being fixed.
